research/engine/pegasus/pegasus/src/lib.rs

```diff
@@ -183,13 +183,16 @@ pub fn startup_with<D: ServerDetect + 'static>(conf: Configuration, detect: D) -
         return Err(StartupError::AlreadyStarted(id));
     }
 
-    if let Some(net_conf) = conf.network_config() {
+    let res =  if let Some(net_conf) = conf.network_config() {
         let addr = net_conf.local_addr()?;
         let conn_conf = net_conf.get_connection_param();
         let addr = pegasus_network::start_up(server_id, conn_conf, addr, detect)?;
         info!("server {} start on {:?}", server_id, addr);
-    }
-    Ok(())
+        Some(addr)
+    } else {
+        None
+    };
+    Ok(res)
```

Comment:
Directly return `Ok(if let Some(...) {} else { ... })`, instead of `let res = if let ...; Ok(res)`

Reply:
Done